acl/src/main/java/org/springframework/security/acls/jdbc/AclClassIdUtils.java

```diff
@@ -124,6 +131,31 @@ private boolean isString(Serializable object) {
 	}
 
 	public void setConversionService(ConversionService conversionService) {
+		Assert.notNull(conversionService, "conversionService must not be null");
 		this.conversionService = conversionService;
 	}
+
+	private static class StringToLongConverter implements Converter<String, Long> {
+		@Override
+		public Long convert(String identifierAsString) {
+			if (identifierAsString == null) {
+				throw new ConversionFailedException(TypeDescriptor.valueOf(String.class),
+						TypeDescriptor.valueOf(Long.class), null, null);
+
+			}
+			return Long.parseLong(identifierAsString);
+		}
+	}
+
+	private static class StringToUUIDConverter implements Converter<String, UUID> {
+		@Override
+		public UUID convert(String identifierAsString) {
+			if (identifierAsString == null) {
+				throw new ConversionFailedException(TypeDescriptor.valueOf(String.class),
+						TypeDescriptor.valueOf(UUID.class), null, null);
+
+			}
+			return UUID.fromString(identifierAsString);
```

Comment:
@jzheaux 

This is only called in case the custom ConversionService passed in via `SetConversionService` does not support StringToLong conversion => should this case be supported?

@see test case `shouldReturnLongEvenIfCustomConversionServiceDoesNotSupportLongConversion`